packages/kbn-monaco/src/esql/language.ts

```diff
@@ -107,6 +111,41 @@ export const ESQLLang: CustomLangModuleType<ESQLCallbacks> = {
           suggestions: wrapAsMonacoSuggestions(suggestions),
         };
       },
+      async resolveCompletionItem(item, token): Promise<monaco.languages.CompletionItem> {
+        if (!callbacks?.getFieldsMetadata) return item;
+        const fieldsMetadataClient = await callbacks?.getFieldsMetadata();
+        const fullEcsMetadataList = EcsMetadataCache.getInstance().getMetadata();
+
+        if (!fullEcsMetadataList || !fieldsMetadataClient || typeof item.label !== 'string')
+          return item;
+
+        const strippedFieldName = removeKeywordSuffix(item.label);
+
+        if (
+          // If item is not a ECS field, no need to fetch metadata
+          item.kind === 4 &&
```

Comment:
I don't understand this check. What kind 4 means?

Reply:
It's just detecting if it's the right completion item kind for the field. I updated it here for some clarity [`fb76609` (#192646)](https://github.com/elastic/kibana/pull/192646/commits/fb766092b4386642259ebd0f6ac95c4ebabb87b2)
'Variable' is what we use currently for fields.

Reply:
Awesome!

Reply:
Do we really need this check? If so, I would suggest we assign `4` to a constant that is used both here and where the field suggestions are generated so that we know that they need to stay in sync.